### ISSUE TITLE
PyQt5 modules linking to Qt

### ISSUE THREAD

**Issue by 3nids** [qgis/QGIS-Mac-Packager#52]
PyQt5 are not pointing to Qt location.

Here is what I used to tackle this

```
for pylib in /opt/QGIS/qgis-deps-0.4.0/stage/lib/python3.7/site-packages/PyQt5/*.so; do
   py_base_lib_name=$(echo $pylib | cut -d/ -f10 | cut -d. -f1)
   echo $py_base_lib_name

   for qt_lib in /opt/Qt/5.14.2/lib/Qt*.framework; do
      qt_base_lib=$(echo $qt_lib | cut -d/ -f6 | cut -d. -f1)
      # echo "install_name_tool -change @rpath/${qt_base_lib}.framework/Versions/5/${qt_base_lib} /opt/Qt/5.14.2/lib/${qt_base_lib}.framework/Versions/Current/${qt_base_lib} /opt/QGIS/qgis-deps-0.4.0/stage/lib/python3.7/site-packages/PyQt5/${py_base_lib_name}.so"
      install_name_tool -change @rpath/${qt_base_lib}.framework/Versions/5/${qt_base_lib} /opt/Qt/5.14.2/lib/${qt_base_lib}.framework/Versions/Current/${qt_base_lib} /opt/QGIS/qgis-deps-0.4.0/stage/lib/python3.7/site-packages/PyQt5/${py_base_lib_name}.so
   done
done
```

**Comment by 3nids** [comment]
I guess this is not the proper approach as this takes the absolute path on my computer.

**Comment by PeterPetrik** [comment]
so where they point? the right approach is that they point to `/opt/Qt/5.14.2/lib/` absolute path, since we assume that the qgis-deps use absolute paths. The @rpath is then used in the bundle step

**Comment by 3nids** [comment]
I miss a bit of understanding how rpath is working.
I switched from `@rpath/QtCore.framework/Versions/5/QtCore` to `/opt/Qt/5.14.2/lib/QtCore.framework/Versions/Current/QtCore` in all PyQt libaries. Does it sound correct?
Otherwise, you cannot build the bindings.

**Comment by PeterPetrik** [comment]
yep

binary can have list of rpaths (install_name_change -add_rpath). and basically @rpath is replaced by one of the paths in the list...

**Comment by PeterPetrik** [comment]
so for example Contents/MacOS/QGIS has list of rpaths. and when it loads libgdal.dylib, it tries all its rpaths to find other libs

**Comment by 3nids** [comment]
but the issue was happening when building.
so is this a matter of adding rpath to PyQt libs pointing to Qt installation?

**Comment by PeterPetrik** [comment]
probably,...

you are building QGIS, or qgis-deps directly? also best to download version 0.4.1 with the latest changes

**Comment by 3nids** [comment]
> you are building QGIS

yes, using deps 0.4.0

**Comment by PeterPetrik** [comment]
it may be fixed in 0.4.1, as i just sucessfully built qgis on out qgis-mac-server... However this annoying bug with libraries signatures/notarization drives me crazy...

**Comment by PeterPetrik** [comment]
```
$ ls /opt/Qt/5.14.2/
android   clang_64  ios       sha1s.txt
$ otool -l /opt/QGIS/qgis-deps-0.5.1/stage/lib/python3.7/site-packages/PyQt5/QtCore.so | grep RPATH -A3
          cmd LC_RPATH
      cmdsize 48
         path @executable_path/../Frameworks (offset 12)
Load command 15
--
          cmd LC_RPATH
      cmdsize 40
         path /opt/Qt/5.14.2/clang_64/lib (offset 12)
Load command 16
```
